lib/Feature/FeatureModelParser.cpp

```diff
@@ -553,6 +553,21 @@ bool FeatureModelSxfmParser::parseFeatureTree(xmlNode *FeatureTree) {
         Name = Name.substr(0, Name.find('(', 0));
       }
 
+      // Search for identifier
+      std::regex Regex(R"(\([a-zA-Z\d_]*\))");
+      std::string CurrentLine = ToStringRef.str();
+      for (std::sregex_iterator Match = std::sregex_iterator(
+               CurrentLine.begin(), CurrentLine.end(), Regex);
+           Match != std::sregex_iterator(); Match++) {
+        std::string Identifier = (*Match).str();
+        Identifier = Identifier.substr(1, Identifier.size() - 2);
+        IdentifierMap[Identifier] = Name;
+        if (Identifier != Name && Name.find(Identifier) != std::string::npos) {
+          llvm::errs() << "Name must not contain ID:'" << Name << "'\n";
+          return false;
+        }
```

Comment:
Why do you assign this before checking?
If we would do the check before, we could move the name string into the map.

Reply:
The issue is that I am using the variable `Name` later for creating a feature with this name using the `FeatureModelBuilder`, so `std::move` is no option here. 
But, of course, I could move the check a line earlier.

Reply:
kk